### PR TITLE
[spirv] Remove BufferBlock decoration (deprecated after SPIRV 1.3).

### PR DESCRIPTION
**Pull request by ehsannas**

### CODE REVIEW

File: tools/clang/lib/SPIRV/RemoveBufferBlockVisitor.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    if (!isBufferBlockDecorationDeprecated(spvOptions))
+      return false;
+
+  return true;
+}
+
+bool RemoveBufferBlockVisitor::visitInstruction(SpirvInstruction *inst) {
+  if (!inst->getResultType())
+    return true;
+
```
Comment (jaebaek) [inline tools/clang/lib/SPIRV/RemoveBufferBlockVisitor.cpp]: So .. are `SpirvAccessChain` and `SpirvPointerType` only cases that require update?

Comment (ehsannas) [reply tools/clang/lib/SPIRV/RemoveBufferBlockVisitor.cpp]: I believe so. OpVariable (and any other operation with a pointer result type) should be fixed by this. For `OpImageTexelPointer`, the storage class of the pointer is `Image`, so no change is needed for it.
And this code ensures that OpAccessChain generates correct storage class for its `base` (even if its base is another OpAccessChain).

Comment (ehsannas) [reply tools/clang/lib/SPIRV/RemoveBufferBlockVisitor.cpp]: for example: 
https://github.com/microsoft/DirectXShaderCompiler/blob/3e5ab5c56b92990f9b45acc0d81df49bd7285707/tools/clang/test/CodeGenSPIRV/vk.1p2.block-decoration.hlsl#L31-L32

Comment (jaebaek) [reply tools/clang/lib/SPIRV/RemoveBufferBlockVisitor.cpp]: Got it.

**Pull request merged**